octree/include/pcl/octree/octree_iterator.h

```diff
@@ -524,6 +523,77 @@ namespace pcl
       };
 
     //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    /** \brief @b Octree iterator class
+     * \note This class implements a forward iterator for traversing octrees in a breadth-first manner.
+     *       It only returns the node at a given depth in the octree, if any.
+     * \ingroup octree
+     * \author Fabien Rozar (fabien.rozar@gmail.com)
+     */
+    template<typename OctreeT>
+      class OctreeBreadthFirstAtDepthIterator : public OctreeBreadthFirstIterator<OctreeT>
+      {
+      public:
+
+        // public typedefs
+        typedef typename OctreeBreadthFirstIterator<OctreeT>::BranchNode BranchNode;
+        typedef typename OctreeBreadthFirstIterator<OctreeT>::LeafNode LeafNode;
+
+        /** \brief Empty constructor.
+          * \param[in] depth_arg Depth level during traversal
+          */
+        explicit
+        OctreeBreadthFirstAtDepthIterator (unsigned int depth_arg = 0);
+
+        /** \brief Constructor.
+        * \param[in] octree_arg Octree to be iterated. Initially the iterator is set to its root node.
+        * \param[in] depth_arg Depth level during traversal
+        */
+        explicit
+        OctreeBreadthFirstAtDepthIterator (OctreeT* octree_arg, unsigned int depth_arg = 0);
+
+        /** \brief Empty deconstructor. */
```

Comment:
This is a pointless docstring, and also not correct, because this is called destructor, not deconstructor.
(Yes, I know you copy-pasted this, but same as the previous comment, we want the new code to be good.)